### DISCUSSION TITLE
Failed to save config for Scarlett 18i8

### DISCUSSION THREAD

**Discussion by ramjr2012** [geoffreybennett/alsa-scarlett-gui#121] [Q&A]
I have just installed **alsa-scarlett-gui** and decided to save the configuration using _File>Save Configuration_.  It failed with the error message in the attached image file.
![ConfigSaveError](https://github.com/geoffreybennett/alsa-scarlett-gui/assets/1427560/715ead65-b33c-4eba-aad7-58bf67d44cb8)
Running the **alsactl** command manually resulted in the following:
```
    $ alsactl store hw:2 -f /run/user/1000/doc/5eefaea0/InitialConfig.state
    alsactl: save_state:1611: Cannot find soundcard 'hw:2'...
```
Running "arecord -l" shows the following:
```
    $ arecord -l
    **** List of CAPTURE Hardware Devices ****
    card 1: Generic [HD-Audio Generic], device 0: ALC233 Analog [ALC233 Analog]
      Subdevices: 1/1
      Subdevice #0: subdevice #0
    card 2: USB [Scarlett 18i8 USB], device 0: USB Audio [USB Audio]
      Subdevices: 0/1
      Subdevice #0: subdevice #0
```
Is this a bug **alsa-scarlett-gui** and if so is this sufficient for a report or do I need to create a bug somewhere?  Or is it an **alsa** bug that needs to be reported to the alsa developers?
Thanks.

**Comment by geoffreybennett** [comment]
Hi @ramjr2012,

What distro and version are you using, and how did you install alsa-scarlett-gui? The error message in your screenshot makes me think it's a packaging error.

As to why running `alsactl store` manually doesn't work, that seems a problem on your system. What does `alsactl --version` say? Can you run `alsa-info.sh` and upload the "ALSA infomation" file that it generates?

Thanks,
Geoffrey.

**Comment by geoffreybennett** [comment]
Something else to try: leave out the `hw:` from the `alsactl store` line or use the other name for it; both of those also work for me:
```
$ aplay -l|grep Scarlett
card 4: USB_2 [Scarlett 18i8 USB], device 0: USB Audio [USB Audio]
$ alsactl store 4 -f test1.state
$ alsactl store USB_2 -f test2.state
$ head -2 test?.state
==> test1.state <==
state.USB_2 {
        control.1 {

==> test2.state <==
state.USB_2 {
        control.1 {
```
